### PR TITLE
{174734955}: Close file descriptor after processing file

### PR DESCRIPTION
**Pull request by morgando**
The `comdb2_files` system table implementation calls `read_write_file` to write a chunk of a file into a buffer. `read_write_file` opens a file but does not close it when it is finished. The changes in this PR add appropriate cleanup to `read_write_file`.

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>3/523 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>queuedb_rollover_noroll1_generated<br>queuedb_rollover<br>truncatesc_ufid_on_generated</p>

File: archive/ar_wrap.c [hunk 1]
```diff
@@ -400,5 +414,10 @@ int read_write_file(dbfile_info *f, void *writer_ctx, writer_cb writer)
 
     f->offset = lseek(fd, 0, SEEK_CUR);
 
-    return 0;
+done:
+    if (fd) {
```
Comment (riverszhang89) [inline archive/ar_wrap.c:418]: A zero fd is valid. Though in practice the code is absolutely fine, `if (fd != -1)` or `if (fd >= 0)` is more idiomatic.

Comment (morgando) [reply archive/ar_wrap.c]: Thanks for catching this! Updated to use `if (fd != -1)`.

**Review by riverszhang89 — APPROVED** [review]

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>7/523 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_truncate<br>truncatesc<br>sicountbug2<br>queuedb_rollover_noroll1_generated<br>queuedb_rollover<br>autoanalyze<br>truncatesc_offline_generated</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>2/524 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>queuedb_rollover<br>truncatesc_offline_generated</p>

**Pull request merged**